broker/lib/jobs/BnRStatusPollerJob.js

```diff
@@ -126,7 +136,117 @@ class BnRStatusPollerJob extends BaseJob {
         Promise.resolve(operationStatusResponse)
       );
   }
+
+  static checkOperationCompletionStatus20(job_data) {
+    logger.info('Checking Operation Completion Status for :', job_data);
+    const operationName = job_data.operation;
+    const instanceInfo = job_data.operation_details;
+    const instance_guid = instanceInfo.instance_guid;
+    const backup_guid = instanceInfo.backup_guid;
+    const deployment = instanceInfo.deployment;
+    const plan = catalog.getPlan(instanceInfo.plan_id);
+    return Promise.try(() => {
+        if (operationName === 'backup') {
+          return lockManager.lock(instance_guid, {
+              lockType: CONST.ETCD.LOCK_TYPE.READ,
+              lockedResourceDetails: {
+                resourceType: CONST.APISERVER.RESOURCE_TYPES.BACKUP,
+                resourceName: CONST.APISERVER.RESOURCE_NAMES.DEFAULT_BACKUP,
+                resourceId: backup_guid,
+                operation: CONST.OPERATION_TYPE.BACKUP
+              }
+            })
+            .catch(err => {
+              if (err instanceof EtcdLockError) {
+                logger.info(`Proceeding as lock is already acquired for the resource: ${instance_guid}`);
+              } else {
+                throw err;
+              }
+            })
+            .then(() => {
+              return BackupManager.createManager(plan)
+                .then(backupManager => backupManager.getServiceFabrikOperationState('backup', instanceInfo));
+            });
+        }
+      })
+      .then(operationStatusResponse => {
```

Comment:
Can you think of reducing the size of this function a bit? I can understand the promise chain being too elongated but can this be moved into smaller chunks? Just think over it